extelasticache/common_elasticache.go

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2024 Steadybit GmbH
+
+package extelasticache
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/service/elasticache"
+	"github.com/steadybit/action-kit/go/action_kit_api/v2"
+	"github.com/steadybit/extension-aws/utils"
+	"github.com/steadybit/extension-kit/extutil"
+)
+
+const (
+	elasticacheNodeGroupTargetId = "com.steadybit.extension_aws.elasticache.node-group"
+)
+
+type ElasticacheClusterAttackState struct {
+	ReplicationGroupID string
+	NodeGroupID        string
+	Account            string
+}
+
+type ElasticacheApi interface {
+	TestFailover(ctx context.Context, params *elasticache.TestFailoverInput, optFns ...func(*elasticache.Options)) (*elasticache.TestFailoverOutput, error)
+	DescribeReplicationGroups(ctx context.Context, params *elasticache.DescribeReplicationGroupsInput, optFns ...func(*elasticache.Options)) (*elasticache.DescribeReplicationGroupsOutput, error)
+}
+
+func convertClusterAttackState(request action_kit_api.PrepareActionRequestBody, state *ElasticacheClusterAttackState) error {
+	state.Account = extutil.MustHaveValue(request.Target.Attributes, "aws.account")[0]
+	state.ReplicationGroupID = extutil.MustHaveValue(request.Target.Attributes, "aws.elasticache.replication-group.id")[0]
+	state.NodeGroupID = extutil.MustHaveValue(request.Target.Attributes, "aws.elasticache.replication-group.node-group.id")[0]
+	return nil
```

Comment:
Always returning `nil` as `error` seems strange? Remove the return type or even inline the function?

Reply:
true, removed the method and put the MustHave things where the method was called